docs/dev_plan.md

```diff
@@ -2,6 +2,110 @@
+wind energy community for land-based and offshore wind turbines. OpenTurbine
+will provide high-fidelity, highly performant structural dynamics models that
+can couple with low-fidelity aerodynamic/hydrodynamic models like those in
+OpenFAST, and high-fidelity computational fluid dynamics (CFD) models like
+those in the WETO and Office of Science supported ExaWind code suite.
+OpenTurbine will be designed to address shortcomings of wind turbine structural
+models and codes that are so important to the success of WETO modeling efforts.
+
+The computational
+algorithms will incorporate robust open-source libraries for mathematical
+operations, resource allocation, and data management. 
+
+## Development priorities and use cases
+
+*Development priorities:* Considering lessons learned from nearly a decade of
+OpenFAST development, OpenTurbine will follow modern software development best
+practices. The development process will require test-driven development,
+version control, hierarchical automated testing, and continuous integration
+leading to a robust development environment. The core data structures will be
+memory efficient and will enable vectorization and parallelization at multiple
+levels. They will be data-oriented in order exploit methods for accelerated
+computing including high utilization of chip resources (e.g., single
+instruction multiple data [SIMD]), parallelization through GPU or other
+hardware, and support for memory-efficient architectures such as the ARM-based
+Apple M-series chips. 
+
+*Use-case priority:* Time-domain simulation of land-based wind turbine dynamics coupled to computational fluid dynamics for fluid-structure-interaction.
+
+## Programming language and models
+
+OpenTurbine is envisioned with a core written in C++ and leveraging Kokkos as its performance-portability library with inspiration from the ExaWind stack including Nalu-Wind CFD.
+
+## High-level design
+
+
+
+
+## Application Programming Interface (API)
+
+The primary goal of the API is to provide data structures and interfaces
+necessary for coupling OpenTurbine to the ExaWind CFD codes for
+fluid-structure-interaction simulations. For land-based wind, the interface
+will be designed to couple the beam finite element models and point-mass
+elements (e.g., representation of the nacelle) to a CFD mesh.  We will leverage
+the mesh mapping that was implemented and tested in the ExaWind codes. The
+representation of the turbine geometry is handled within the fluid solver,
+either as a three-dimensional surface mesh for high-fidelity geometry-resolved
+simulations, or as an actuator-line for mid-fidelity simulations. Those
+algorithms will work well for problems where a floating platform is represented
+as a point-mass. For geometry-resolved floating-platform offshore simulations,
+the API will be expanded to handle mapping from the structure surface mesh (if
+solid/shell elements are used) to the CFD mesh.
```

Comment:
Why this caveat for solid/shell elements? Nalu-wind's current interaction with OpenFast maps to/from sides/faces of the surface.

Reply:
@alanw0 -- the mapping right now is from the CFD surface to a line representation of the blade (or maybe even a point representation).  Nonetheless, we'll need a surface-to-surface coupling once shell/solid finite elements are used.